software/deploy/deploy.go

```diff
@@ -195,13 +202,14 @@ func buildPushDockerImage(houstonClient houston.ClientInterface, c *config.Conte
 		remoteImage = fmt.Sprintf("%s/%s", registry, airflow.ImageName(name, nextTag))
 		token = c.Token
 	}
-
 	err = imageHandler.Push(remoteImage, "", token)
 	if err != nil {
 		return err
 	}
-
 	if byoRegistryEnabled {
+		if useShaAsTag {
+			remoteImage = fmt.Sprintf("%s@%s", registry, sha)
```

Comment:
```suggestion
			sha, err := imageHandler.GetBuiltImageSha()
			if (sha == "") || (err != nil) {
				return fmt.Errorf("failed to get image sha: %w", err)
			}
			remoteImage = fmt.Sprintf("%s@%s", registry, sha)
```
nit: keeping usage and value fetching close together to simplify the logic